### PR TITLE
fix: No more stuttering with large item stacks in inv (#35)

### LINKED ISSUE #35: Performance drops when working with large item quantities
When i have the plugin on if i take coins out of my bank or the ge the game will freeze for multiple seconds.

### PR DESCRIPTION
**Pull request by Infinitay**
fix: No more stuttering with large item stacks in inv

# EssencePouchTrackingPlugin
- Iterate through the #elementSet to only loop through the item id once instead of per occurance in #onItemContainerChanged

I made the mistake of thinking that iterating a `Multiset` would only loop the element itself, like a key in a `Map`. However, it will iterate `n` times as as the size of the Multiset - the total # of occurrences of all elements. As a result, there were stutters when withdrawing items with large quantities such as coins.

Fixes #35

### CODE REVIEW

**Comment by Infinitay** [comment]
# EssencePouches
- Refactor class to use static map references for the pouch types by their item ids: one map for regular pouches and another for degraded pouches by item ids
- Refactored #getPouch, #isPouchDegraded, #createPouch to use the maps to fetch the pouches

[9804986](https://github.com/Infinitay/essence-pouch-tracking/pull/41/commits/98049864779173e8ba840ab30c3b0fb5c52f08a2): `New`
[494399f](https://github.com/Infinitay/essence-pouch-tracking/pull/41/commits/494399f36a06aab29d8f50c7ed8900ea380b9c4c): `Old`

image

tldr performance gain albeit miniscule
